### ISSUE TITLE
include!() can silently ignore part of a file

### ISSUE THREAD

**Issue by fstirlitz** [rust-lang/rust#42890]
Suppose you've got a file `a.rs`:

```rust
fn main() {
	include!("b.rs");
}
```

And `b.rs`:
```rust
println!("HAHAHAHAHAHA");
println!("no");
```

`a.rs` compiles with no warnings into an executable that prints just one line of output.

This is quite surprising behaviour. I believe at the very least this should issue a diagnostic of some sort; though preferably I'd have it include the whole file. (I know I can just put braces inside the file, but I'd rather not do that; my original scenario was much more elaborate, and involved invoking macros inside the included file.)

**Comment by qnighy** [comment]
As a reference, #41497 might be related to this.

**Comment by nagisa** [comment]
Yeah, seems a lot like a duplicate.
